StackExchange.Redis/StackExchange/Redis/Interfaces/IDatabase.cs

```diff
@@ -1389,6 +1389,225 @@ public interface IDatabase : IRedis, IDatabaseAsync
+        /// <param name="maxLength">The maximum length of the stream.</param>
+        /// <param name="useApproximateMaxLength">If true, the "~" argument is used to allow the stream to exceed max length by a small number. This improves performance when removing messages.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The ID of the newly created message.</returns>
+        /// <remarks>https://redis.io/commands/xadd</remarks>
+        RedisValue StreamAdd(RedisKey key, NameValueEntry[] streamPairs, string messageId = "*", int? maxLength = null, bool useApproximateMaxLength = false, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Change ownership of messages consumed, but not yet acknowledged, by a different consumer. This method returns the complete message for the claimed message(s).
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="consumerGroup">The consumer group.</param>
+        /// <param name="claimingConsumer">The consumer claiming the given message(s).</param>
+        /// <param name="minIdleTimeInMs">The minimum message idle time to allow the reassignment of the message(s).</param>
+        /// <param name="messageIds">The IDs of the messages to claim for the given consumer.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The messages successfully claimed by the given consumer.</returns>
+        /// <remarks>https://redis.io/topics/streams-intro</remarks>
+        RedisStreamEntry[] StreamClaimMessages(RedisKey key, RedisValue consumerGroup, RedisValue claimingConsumer, long minIdleTimeInMs, RedisValue[] messageIds, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Change ownership of messages consumed, but not yet acknowledged, by a different consumer. This method returns the IDs for the claimed message(s).
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="consumerGroup">The consumer group.</param>
+        /// <param name="claimingConsumer">The consumer claiming the given message(s).</param>
+        /// <param name="minIdleTimeInMs">The minimum message idle time to allow the reassignment of the message(s).</param>
+        /// <param name="messageIds">The IDs of the messages to claim for the given consumer.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The message IDs for the messages successfully claimed by the given consumer.</returns>
+        /// <remarks>https://redis.io/topics/streams-intro</remarks>
+        RedisValue[] StreamClaimMessagesReturningIds(RedisKey key, RedisValue consumerGroup, RedisValue claimingConsumer, long minIdleTimeInMs, RedisValue[] messageIds, CommandFlags flags = CommandFlags.None);
```

Comment:
I was thinking about these, and it finally occurs to me: everything is `Messages`. We need not say it. For example, we don't call it `StreamAddMessage()`, `StreamAcknowledgeMessages()`, etc.

What if we shortened these `StreamClaim()` and `StreamClaimIdsOnly()`? Thoughts?

Reply:
I like it, making those changes now.